streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -233,27 +293,39 @@ private static void decodeVersionTwoData(final AssignmentInfo assignmentInfo,
         return partitions;
     }
 
+    private static void decodeVersionThreeData(final AssignmentInfo assignmentInfo,
+                                               final DataInputStream in) throws IOException {
+        decodeActiveTasks(assignmentInfo, in);
```

Comment:
It seems like these two approaches are fighting for dominance in this code.

Personally, I favor the "flat" design, where these methods don't call each other. It's just easier to read when each one in isolation lists all the operations it needs to do.

It's also more general, since a hypothetical future version N might add some new metadata in the middle, or even drop some previous metadata and so wouldn't be able to call decodeVersion{N-1} in its implementation.

Reply:
Version three will change and be different to version two. Thus, I prefer to not just call encodeVersionTwo. Guess it's personal taste.